dbcon/joblist/groupconcat.cpp

```diff
@@ -1038,59 +1452,61 @@ void GroupConcatNoOrder::processRow(const rowgroup::Row& row)
     fCurrentLength += estLen;
     fRowGroup.incRowCount();
     fRow.nextRow();
+    auto newSize = fRowGroup.getSizeWithStrings(fRowsPerRG);
+    if (newSize > fCurMemSize)
+    {
+      auto diff = newSize - fCurMemSize;
+      fCurMemSize = newSize;
+      fMemSize += diff;
+    }
 
     if (fRowGroup.getRowCount() >= fRowsPerRG)
     {
-      // A "postfix" but accurate RAM accounting that sums up sizes of RGDatas.
-      uint64_t newSize = fRowGroup.getSizeWithStrings();
-
-      if (!fRm->getMemory(newSize, fSessionMemLimit))
-      {
-        cerr << IDBErrorInfo::instance()->errorMsg(fErrorCode) << " @" << __FILE__ << ":" << __LINE__;
-        throw IDBExcept(fErrorCode);
-      }
-      fMemSize += newSize;
-
-      fDataQueue.push(fData);
-      fData.reinit(fRowGroup, fRowsPerRG);
-      fRowGroup.setData(&fData);
-      fRowGroup.resetRowGroup(0);
-      fRowGroup.getRow(0, &fRow);
+      createNewRGData();
     }
   }
 }
 
 void GroupConcatNoOrder::merge(GroupConcator* gc)
 {
-  GroupConcatNoOrder* in = dynamic_cast<GroupConcatNoOrder*>(gc);
+  auto* in = dynamic_cast<GroupConcatNoOrder*>(gc);
+  assert(in != nullptr);
 
-  while (in->fDataQueue.size() > 0)
+  for (auto& i : in->fDataVec)
   {
-    fDataQueue.push(in->fDataQueue.front());
-    in->fDataQueue.pop();
+    fDataVec.emplace_back(std::move(i));
   }
-
-  fDataQueue.push(in->fData);
+  fRowGroup.setData(fDataVec.back().get());
+  fRowGroup.setUseOnlyLongString(true);
+  fRowGroup.initRow(&fRow);
+  fRowGroup.getRow(fRowGroup.getRowCount(), &fRow);
   fMemSize += in->fMemSize;
-  in->fMemSize = 0;
+  fCurMemSize = in->fCurMemSize;
+  in->fMemSize = in->fCurMemSize = 0;
 }
 
 uint8_t* GroupConcatNoOrder::getResultImpl(const string& sep)
 {
   ostringstream oss;
   bool addSep = false;
 
-  fDataQueue.push(fData);
   size_t prevResultSize = 0;
 
   bool isNull = true;
-  while (fDataQueue.size() > 0)
+  bool addBrackets = true;
```

Comment:
If `addBrackets` is always true here than I didn't get why do we have it in the line 1505?

Reply:
We set it to false after adding the opening bracket at line 1510.